Bottom: e87c3f3a23523b4cb6d4b18c0dcfaf1df12b0467
Top:    d84653ca88816c283ee7e2e4cc731ca5d8b36cab
Author: Mark de Wever <koraq@xs4all.nl>
Date:   2021-08-07 14:42:58 +0200

Refresh of toy-with-a-gui

---

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ad4b770..a545078 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -2,6 +2,10 @@ add_subdirectory(modules)
 add_executable(rpn
 	rpn.cpp
 )
+target_compile_options(rpn
+	PRIVATE
+		${module_compile_options}
+)
 target_link_libraries(rpn
 	calculator
 	fltk
diff --git a/src/modules/calculator/CMakeLists.txt b/src/modules/calculator/CMakeLists.txt
index dc073b6..78b61cf 100644
--- a/src/modules/calculator/CMakeLists.txt
+++ b/src/modules/calculator/CMakeLists.txt
@@ -7,10 +7,12 @@ target_compile_options(calculator
 	PRIVATE
 		${module_compile_options}
 )
+add_module(calculator calculator.cpp)
 add_module(calculator.controller controller.cpp)
 add_module(calculator.model model.cpp)
 add_module(calculator.value value.cpp)
 add_dependencies(calculator
+	calculator.pcm
 	calculator.controller.pcm
 	calculator.model.pcm
 	calculator.value.pcm
diff --git a/src/modules/calculator/model.cpp b/src/modules/calculator/model.cpp
index cb0ace1..c188793 100644
--- a/src/modules/calculator/model.cpp
+++ b/src/modules/calculator/model.cpp
@@ -18,7 +18,7 @@ export import calculator.value;
 
 import<string>;
 import<string_view>;
-import<vector>;
+export import<vector>;
 
 namespace calculator {
 
diff --git a/src/rpn.cpp b/src/rpn.cpp
index fe3b0e7..4e95924 100644
--- a/src/rpn.cpp
+++ b/src/rpn.cpp
@@ -12,7 +12,7 @@
  * See the COPYING file for more details.
  */
 
-// import calculator.model;
+ import calculator;
 
 #include <FL/Fl.H>
 #include <FL/Fl_Box.H>
@@ -47,6 +47,8 @@ private:
   Fl_Box input_{5, 210, 300, 20};
 
   std::string buffer_{};
+  calculator::tmodel model_;
+  calculator::tcontroller controller_{model_};
 };
 
 void twindow::process_input_event() noexcept {
